build.xml

```diff
@@ -69,7 +69,6 @@
             <fileset dir="bin/" id="fileset.class">
                 <include name ="**/*.class" />
                 <exclude name="**/test/**/*" />
-                <exclude name="smoothing/*" />
```

Comment:
Um, the idea behind implementing this as an external process was to make the cancel feature work. Cancellation works by invoking `Future.cancel`, which attempts to interrupt the thread that corresponds to that `Future`, which only works if the thread is waiting for something than can throw an `InterruptedException` (e.g. if it's blocked in a [`Process.waitFor`](http://docs.oracle.com/javase/7/docs/api/java/lang/Process.html#waitFor%28%29) call). If you don't like the idea of running a separate smoothing process, you can probably make cancellation work by starting a separate smoothing thread and using `Thread.join` in the main thread.